### PR TITLE
Feat: add config management apis

### PR DESCRIPTION
**Pull request by zzxwill**
Added some APIs for config management

Signed-off-by: Zheng Xi Zhou <zzxwill@gmail.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by barnettZQG — COMMENTED** [review]
1.  Further refine the configuration definition. 
2.  Filter the configuration definitions in the listing component definitions API.

File: charts/vela-minimal/templates/defwithtemplate/config-helm-repository.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        	type: "Opaque"
+
+        	if parameter.https != _|_ {
+        		stringData: parameter.https
+        	}
+        	if parameter.ssh != _|_ {
+        		stringData: parameter.ssh
+        	}
+        }
+        parameter: {
```
Comment (barnettZQG) [inline charts/vela-minimal/templates/defwithtemplate/config-helm-repository.yaml:41]:
```
repoType: oss | https | ssh
oss?: {
        bucket:   string
        endpoint: string
}
https?: {
      url:      string
      username: string
      password: string
}
// +usage=https://fluxcd.io/legacy/helm-operator/helmrelease-guide/chart-sources/#ssh
ssh?: {
    url:      string
    identity: string
}
```

File: pkg/apiserver/rest/usecase/config.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	terraformProviderType = "terraform-provider"
+	// LabelProject is the label to store project name in an application
+	LabelProject = "project"
+)
+
+// ConfigHandler handle CRUD of configs
+type ConfigHandler interface {
+	ListConfigTypes(ctx context.Context, query string) ([]*apis.ConfigType, error)
+	GetConfigType(ctx context.Context, configType string) (*apis.ConfigType, error)
+	CreateConfig(ctx context.Context, req apis.CreateApplicationRequest) (*apis.ApplicationBase, error)
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/config.go:51]: CreateApplicationRequest?  Please redefine the request and response structs.

File: pkg/apiserver/rest/usecase/config.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+	labelVal              = "velacore-config"
+	definitionAlias       = definition.UserPrefix + "alias.config.oam.dev"
+	definitionType        = definition.UserPrefix + "type.config.oam.dev"
+	terraformProviderType = "terraform-provider"
+	// LabelProject is the label to store project name in an application
+	LabelProject = "project"
+)
+
+// ConfigHandler handle CRUD of configs
+type ConfigHandler interface {
```
Comment (barnettZQG) [inline pkg/apiserver/rest/usecase/config.go:48]: Please provide a query API for listing configs inside a specified project.

File: pkg/apiserver/rest/webservice/config.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+	return &configWebService{
+		handler: u,
+	}
+}
+
+type configWebService struct {
+	handler usecase.ConfigHandler
+}
+
+func (s *configWebService) GetWebService() *restful.WebService {
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/config.go:42]: Please add the `authCheckFilter` and `CheckPerm` filters.

File: pkg/apiserver/rest/usecase/config.go (hunk 2 above)
Comment (zzxwill) [reply pkg/apiserver/rest/usecase/config.go]: It's indeed an application request and I will create an application based on the request.

File: charts/vela-minimal/templates/defwithtemplate/config-helm-repository.yaml (hunk 1 above)
Comment (wonderflow) [reply charts/vela-minimal/templates/defwithtemplate/config-helm-repository.yaml:41]: +1 for this

**Review by barnettZQG — CHANGES_REQUESTED** [review]
Please change the code reference the comment.

File: pkg/apiserver/rest/usecase/config.go (hunk 3 above)
Comment (zzxwill) [reply pkg/apiserver/rest/usecase/config.go:51]: We are working to validate terraform providers and dex connectors at this very moment. Can this be implemented in next later?

File: pkg/apiserver/rest/webservice/config.go (hunk 4 above)
Comment (zzxwill) [reply pkg/apiserver/rest/webservice/config.go:42]: What are they?

File: charts/vela-minimal/templates/defwithtemplate/config-helm-repository.yaml (hunk 1 above)
Comment (zzxwill) [reply charts/vela-minimal/templates/defwithtemplate/config-helm-repository.yaml]: Removed Helm Chart repository definition.

File: pkg/apiserver/rest/webservice/config.go (hunk 4 above)
Comment (barnettZQG) [reply pkg/apiserver/rest/webservice/config.go:42]: Ref:
image

Comment (barnettZQG) [reply pkg/apiserver/rest/webservice/config.go:42]: For resource name definition:

in usecase/rbac.go:L210 
```
"config_types": {
		pathName: "configType",
		subResources: map[string]resourceMetadata{
			"config": {pathName: "configName"},
		},
	},
```

File: pkg/apiserver/rest/usecase/config.go [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+	"github.com/oam-dev/kubevela/apis/types"
+	"github.com/oam-dev/kubevela/pkg/apiserver/clients"
+	apis "github.com/oam-dev/kubevela/pkg/apiserver/rest/apis/v1"
+	"github.com/oam-dev/kubevela/pkg/definition"
+	"github.com/oam-dev/kubevela/pkg/utils/apply"
+)
+
+const (
+	definitionAlias       = definition.UserPrefix + "alias.config.oam.dev"
+	definitionType        = definition.UserPrefix + "type.config.oam.dev"
```
Comment (wonderflow) [inline pkg/apiserver/rest/usecase/config.go:45]: 这个上次结论是要特判去掉

File: pkg/apiserver/rest/usecase/config.go [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+	t := &apis.ConfigType{
+		Alias:       d.Annotations[definitionAlias],
+		Name:        configType,
+		Description: d.Annotations[types.AnnoDefinitionDescription],
+	}
+	return t, nil
+}
+
+func (u *defaultConfigHandler) CreateConfig(ctx context.Context, req apis.CreateApplicationRequest) error {
+	var env = "Unknown"
```
[the remark this answers is not in the document]
Comment (wonderflow) [reply pkg/apiserver/rest/usecase/config.go:147]: note the mark

File: pkg/apiserver/rest/usecase/config.go [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+
+	t := &apis.ConfigType{
+		Alias:       d.Annotations[definitionAlias],
+		Name:        configType,
+		Description: d.Annotations[types.AnnoDefinitionDescription],
+	}
+	return t, nil
+}
+
+func (u *defaultConfigHandler) CreateConfig(ctx context.Context, req apis.CreateApplicationRequest) error {
```
Comment (wonderflow) [inline pkg/apiserver/rest/usecase/config.go:146]: why using CreateApplicationRequest in create config API?

File: pkg/apiserver/rest/usecase/config.go (hunk 5 above)
Comment (zzxwill) [reply pkg/apiserver/rest/usecase/config.go:45]: @Somefive Can you implement this request?

**Pull request closed without merging**